deployment/openshift/form_flow_files_dc.yaml

```diff
@@ -0,0 +1,232 @@
+kind: Template
+apiVersion: template.openshift.io/v1
+metadata:
+  name: forms-flow-files
+  tags: forms-flow-files
+
+parameters:
+  - name: NAME
+    displayName: App Name
+    description: The name of the application
+    required: true
+    value: forms-flow-files
+
+  - name: BACKUP_SUFFIX
+    displayName: Backup Prefix
+    description: The prefix added to the backup
+    required: true
+    value: backup
+
+  - name: PORT
+    displayName: App Port
+    required: true
+    value: 4100
+  
+  - name: FILES_MAX_UPLOAD_SIZE
+    displayName: Max File Upload Size
+    required: true
+    value: 32mb
+  
+  - name: FILES_PROVIDERS
+    displayName: File Providers
+    required: true
+    value: file
+
+  - name: FILES_UPLOAD_DIR
+    displayName: Files Upload Directory
+    required: true
+    value: /uploads
+
+  - name: HOST_URL
+    displayName: App Host URL
+    required: true
+    value: digital-journeys-files.apps.silver.devops.gov.bc.ca
+  
+  - name: CPU_REQUEST
+    displayName: Resources CPU Request
+    description: The resources CPU request (in cores) for this build.
+    required: true
+    value: 300m
+  
+  - name: CPU_LIMIT
+    displayName: Resources CPU Limit
+    description: The resources CPU limit (in cores) for this build.
+    required: true
+    value: 700m
+
+  - name: MEMORY_REQUEST
+    displayName: Resources Memory Request
+    description: The resources Memory request (in Mi, Gi, etc) for this build.
+    required: true
+    value: 512M
+
+  - name: MEMORY_LIMIT
+    displayName: Resources Memory Limit
+    description: The resources Memory limit (in Mi, Gi, etc) for this build.
+    required: true
+    value: 2G
+  
+  - name: REPLICAS
+    displayName: REPLICAS
+    description: The number of statefulSet replicas to use.
+    required: true
+    value: '1'
+
+  - name: IMAGE_BASE_URL
+    display: Image Base URL
+    description: Base Image URL or repository URL of the deployed image
+    required: true
+    value: image-registry.openshift-image-registry.svc:5000
+
+  - name: NAMESPACE
+    displayName: Namespace
+    description: Namespace of the deployment
+    required: true
+
+  - name: IMAGE_TAG
+    displayName: Image App Tag
+    description: Tag of the application environment being deployed
+    required: true
+
+objects:
+  - kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: ${NAME}-data
+      labels:
+        app: ${NAME}
+    spec:
+      accessModes:
+        - ReadWriteMany               # this is ReadWriteOnce currently
```

Comment:
I'm wondering what will happen if we change this from `ReadWriteOnce` to `ReadWriteMany` as I believe it is immutable?
Will it replace the old volume and all the data within it? As we have to make sure that we retain the data that's already there

Reply:
Great catch! I had the same question and thought of checking this with you, but it got skipped completely. I added it for deployment failures I was facing while modifying pod definitions.

> Will it delete the underlying persistent volume as well? I am not very sure.

With that in my mind, let me change it back to RWO (as in console currently) and change it later, if needed.

Reply:
Sounds good! Yeah we would want to change it to RWM fairly soon so we're not limited to just one pod, but sounds good to get this merged first